tests/endpoints/echo/server/app.go

```diff
@@ -120,6 +134,51 @@ func main() {
 	log.Fatal(err)
 }
 
+// RstListener is a listener that sends an RST when closed instead of a FIN.
+type RstListener struct {
+	l net.Listener
+}
+
+func NewRstListener(addr string) (*RstListener, error) {
+	l, err := net.Listen("tcp", addr)
+	return &RstListener{l: l}, err
+}
+
+// Accept is like a normal Accept except that it ses SO_LINGER time to 0, so that when
+// close is called on the socket it will send an RST instead of a FIN.
+func (l *RstListener) Accept() (net.Conn, error) {
+	conn, err := l.l.Accept()
+	if err != nil {
+		return nil, err
+	}
+
+	tcpConn, ok := conn.(*net.TCPConn)
+	if !ok {
+		return nil, fmt.Errorf("Expected a TCP connection")
+	}
+
+	err = tcpConn.SetLinger(0)
+	if err != nil {
+		fmt.Printf("get error before sending rst: %v", err)
+	}
+	return conn, err
+}
+
+func (l *RstListener) Close() error {
+	return l.l.Close()
+}
+
+func RejectMiddleWare(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if *rejectRequestNum > 0 {
+			*rejectRequestNum -= 1
```

Comment:
Do we need a mutex here? I assume it's fine, our backends don't receive parallel requests.

Reply:
ACK. We are sending sequential requests so it should be fine. Would be easy to add if needed.